force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +121,23 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
+        Integer result = 0;
         /*
         for () {
             // get individual characters from the string - hint use substring or split
             // check if the character is a vowel
             // if it is a vowel, increment the result Integer
         }
         */
+        for(Integer i = 0 ; i < str.length(); i++){
+            // (startIndex,endIndex)
+            String character = str.substring(i, i +1);
+         if(character.equals('a') || character.equals('e') || character.equals('i') ||character.equals('o') ||character.equals('u')){
+            result++;
```

Comment:
This works but  you are creating a lot of Stings in memory you don't really need. 

When checking for membership I think a Set is almost always the preferred data structure. 
```
 public static Integer countVowels(String str) {
        // Initialize the result Integer
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
    ```